### PR TITLE
LineageRegistrationPlugin: add copy spot labels functionality

### PR DESCRIPTION
**Pull request by maarzt**
issue #60

### CODE REVIEW

File: src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java [hunk 1]
```diff
@@ -85,6 +85,14 @@ public class LineageRegistrationFrame extends JFrame
 			+ "</ul>"
 			+ "</body></html>";
 
+	private static final String COPY_LABELS_TOOLTIP = "<html><body>"
+			+ "Use the found correspondences to copy spot labels from one project to the other.<br>"
+			+ "<br>"
+			+ "The correspondences are on a level of cells / branches.<br>"
+			+ "That is why the all the spots that belong to a cell / branch will get the same label.<br>"
+			+ "Labels on individual spot can not copied."
```
Comment (stefanhahmann) [inline src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java:93]:
```suggestion
			+ "Labels of individual spot can not copied."
```

File: src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java [hunk 2]
```diff
@@ -85,6 +85,14 @@ public class LineageRegistrationFrame extends JFrame
 			+ "</ul>"
 			+ "</body></html>";
 
+	private static final String COPY_LABELS_TOOLTIP = "<html><body>"
+			+ "Use the found correspondences to copy spot labels from one project to the other.<br>"
+			+ "<br>"
+			+ "The correspondences are on a level of cells / branches.<br>"
```
Comment (stefanhahmann) [inline src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java:91]:
```suggestion
			+ "The correspondences are on the level of cells / branches.<br>"
```

**Review by stefanhahmann — CHANGES_REQUESTED** [review]
I tested the function and it mainly worked as expected with one inconsistency:

Currently, the branch graph shows the label of the last spot that belongs to it, making the label of the *last* spot the significant label of the branch. This copy label function uses the label of the *first* spot to copy it to all spots of equivalent branch in the other project. This is okay, as long as users only work on the branch graph, but may lead to inconsistent / unexpected behavior, if the users use both, the branch graph and the spot graph.

I recommend some action to overcome this inconsistency.

File: src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationUtils.java [hunk 3]
```diff
@@ -318,4 +318,28 @@ public static void plotAngleAgainstTimepoint( RefDoubleMap< Spot > angles )
...
+
+	public static void copySpotLabelsFromAtoB( final RegisteredGraphs registration )
+	{
+		final ModelGraph graphB = registration.graphB;
+		final Spot refB = graphB.vertexRef();
+		try
+		{
+			for ( final Spot spotA : registration.mapAB.keySet() )
+			{
+				boolean hasLabel = !Integer.toString( spotA.getInternalPoolIndex() ).equals( spotA.getLabel() );
```
[the remark this answers is not in the document]
Comment (maarzt) [reply src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationUtils.java:330]: I opened a PR https://github.com/mastodon-sc/mastodon/pull/318 that addresses this suggestion.

File: src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java [hunk 4]
```diff
@@ -176,9 +184,12 @@ public LineageRegistrationFrame( Listener listener )
 		add( new JLabel( "Sort TrackScheme:" ) );
 		add( newOperationButton( "project A", SORT_TRACKSCHEME_TOOLTIP, listener::onSortTrackSchemeAClicked ), "split 2" );
 		add( newOperationButton( "project B", SORT_TRACKSCHEME_TOOLTIP, listener::onSortTrackSchemeBClicked ), "wrap" );
+		add( new JLabel( "Copy spot labels:" ) );
```
[the remark this answers is not in the document]
Comment (maarzt) [reply src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java:187]: The text says now "copy spot labels"

File: src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java (hunk 1 above)
Comment (maarzt) [reply src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java:93]: It's corrected.

File: src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java (hunk 2 above)
Comment (maarzt) [reply src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationFrame.java:91]: Thank you for spotting this.

File: src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationUtils.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (stefanhahmann) [reply src/main/java/org/mastodon/mamut/tomancak/lineage_registration/LineageRegistrationUtils.java:330]: @maarzt https://github.com/mastodon-sc/mastodon/pull/318 has been merged in the meantime. Thus you could update here.

**Pull request merged**